core/src/com/mygdx/minigolf/controller/systems/ObjectiveSystem.java

```diff
@@ -0,0 +1,52 @@
+package com.mygdx.minigolf.controller.systems;
+
+import com.badlogic.ashley.core.ComponentMapper;
+import com.badlogic.ashley.core.Engine;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.EntitySystem;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.mygdx.minigolf.model.components.Objective;
+import com.mygdx.minigolf.model.components.Physical;
+import com.mygdx.minigolf.model.components.Player;
+
+// All logic should go in the system
+public class ObjectiveSystem extends EntitySystem{
+    private ImmutableArray<Entity> objectives;
+    private ImmutableArray<Entity> players;
+
+    private ComponentMapper<Physical> physicalMapper;
+    private ComponentMapper<Objective> objectiveMapper;
+    private ComponentMapper<Player> playerMapper;
+
+    public ObjectiveSystem(){
+        physicalMapper = ComponentMapper.getFor(Physical.class);
+        objectiveMapper = ComponentMapper.getFor(Objective.class);
+        playerMapper = ComponentMapper.getFor(Player.class);
+    }
```

Comment:
Dette er mer et kodestil "problem" enn no annet, men vi burde bli enige om om vi skal instansiere attributter utenfor constructoren eller ikke, når det ikke er noen logikk bundet til instansieringen . I dette tilfellet for eksempel er det ingen logikk involvert i instansieringen, så man kunne like gjerne skrevet 
```
    private ComponentMapper<Physical> physicalMapper = ComponentMapper.getFor(Physical.class);
    private ComponentMapper<Objective> objectiveMapper = ComponentMapper.getFor(Objective.class);
    private ComponentMapper<Player> playerMapper = ComponentMapper.getFor(Player.class);
```
øverst i klassen. Noen foretrekker å gjøre det slik, og noen foretrekker å ha all instansiering i konstrutøren. Det viktigste er kanskje at vi blir enige om hva vi vil gjøre.